source/fundamentals/typescript.txt

```diff
@@ -0,0 +1,97 @@
+==========
+TypeScript
+==========
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecols
+
+Overview
+--------
+
+In this guide, you can learn about the **TypeScript** specific features and limitiations
+of the MongoDB Node.js driver. TypeScript is a strongly typed programming
+language that compiles to JavaScript.
+
+All TypeScript features of the driver are optional, and all valid JavaScript
+code written with the driver is also valid TypeScript code. 
+
+For more information on TypeScript, see 
+`the official TypeScript website <https://www.typescriptlang.org/>`__.
+
+Features
+--------
+
+If you use TypeScript, you can specify a type for some classes in the driver.
+All classes that accept a type paramater in the driver have the default type
+:node-api-4.0:`Document <interfaces/document.html>`.
+
+The following classes accept any type that extends the ``Document``
+interface:
+
+- :node-api-4.0:`Collection <classes/collection.html>`
+- :node-api-4.0:`ChangeStream <classes/changestream.html>`
+
+You can pass a type parameter that extends the ``Document`` interface like this:
```

Comment:
Suggestion: "To pass a type ... interface:"

Reply:
We use the "you can pass... like this:" style for introducing code snippets in drivers documentation pretty frequently and it think it is good for talking about something that demonstrates the look and feel of a feature that could be used in lots of different ways. 

"To pass a type... interface" feels a little too restrictive, as the code snippet is not the be all end all way to do this, just highlights some of the syntax. Open to further discussion if you disagree!